src/components/MailboxPicker.vue

```diff
@@ -0,0 +1,70 @@
+<template>
+	<Treeselect
+		ref="Treeselect"
+		:options="mailboxes"
+		:multiple="false"
+		:clearable="false"
+		v-bind="$attrs"
+		v-on="$listeners" />
+</template>
+<script>
+import Treeselect from '@riophae/vue-treeselect'
+import '@riophae/vue-treeselect/dist/vue-treeselect.css'
+
+export default {
+	name: 'MailboxPicker',
+	components: {
+		Treeselect,
+	},
+	props: {
+		accountid: {
+			type: Number,
+			required: true,
+		},
+	},
+	computed: {
+		mailboxes() {
+			const boxes = []
+			this.getMailboxes(boxes)
+			return boxes
+		},
+	},
+	methods: {
+		getMailboxes(root, folderid) {
+			let folders = []
```

Comment:
```suggestion
			let mailboxes = []
```

for consistency :)

Reply:
We now need 2 arrays here...

Reply:
Not necessarily. Instead of iterating on one array and filling another one (imperative coding) you can *map* the mailboxes to the new data structure and just return the result. No need to store anything a second time. If that's too much to ask then leave it and I can clean up the code later :v:

Reply:
It's definitly not too much to ask, I simply do not understand how to do that. Can you give me a hint?

Reply:
Then I'm sorry for not explaining it.

Currently you do
```js
let mailboxes = ...
const boxLevel = []
mailboxes.forEach((folder) => {
    ...
    boxLevel.push({ ... })
})
return boxLevel
```

but you can simplify to

```js
let mailboxes = ...
return mailboxes.map((mailbox) => {
    return {
        ...,
        children: this.getMailboxes(mailbox.databseId)
    }
})
```

Reply:
Wow, strange syntax ;-) Done!